Gems/Meshlets/Code/Source/Meshlets/MeshletsDispatchItem.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <MeshletsDispatchItem.h>
+
+#include <Atom/RPI.Public/Shader/ShaderResourceGroup.h>
+#include <Atom/RPI.Public/Shader/Shader.h>
+
+#include <limits>
+
+#include "MeshletsAssets.h"
+
+namespace AZ
+{
+    namespace Meshlets
+    {
+        #define MESHLETS_THREAD_GROUP_SIZE 64
+
+        MeshletsDispatchItem::~MeshletsDispatchItem()
+        {
+        }
+
+        void MeshletsDispatchItem::InitDispatch(
+            RPI::Shader* shader,
+            Data::Instance<RPI::ShaderResourceGroup> meshletsDataSrg,
+            uint32_t meshletsAmount )
+        {
+            RHI::DispatchDirect dispatchArgs(
+                meshletsAmount, 1, 1,
+                MESHLETS_THREAD_GROUP_SIZE, 1, 1
+            );
+
+            m_dispatchItem.m_arguments = dispatchArgs;
+            m_dispatchItem.m_shaderResourceGroupCount = 1;      // Per pass srg can be added by the individual passes
+            m_dispatchItem.m_shaderResourceGroups =
+            {
+                meshletsDataSrg->GetRHIShaderResourceGroup()
+            };
+            m_meshletsDataSrg = meshletsDataSrg;    // can also be retrieve directly from the m_dispatchItem
+
+            m_shader = shader;
+            if (shader)
+            {
+                RHI::PipelineStateDescriptorForDispatch pipelineDesc;
+                m_shader->GetVariant(RPI::ShaderAsset::RootShaderVariantStableId).ConfigurePipelineState(pipelineDesc);
+                m_dispatchItem.m_pipelineState = m_shader->AcquirePipelineState(pipelineDesc);
+            }
```

Comment:
```suggestion
            SetPipelineState(shader);
```

Reply:
Not sure about this change and rather not risk it at this stage.  Is this a blocker?